modal/_container_io_manager.py

```diff
@@ -209,9 +208,7 @@ def _init(self, container_args: api_pb2.ContainerArguments, client: _Client):
         self.current_input_id = None
         self.current_input_started_at = None
 
-        self._input_concurrency = None
-
-        self._semaphore = None
+        self._dynamic_semaphore = None
```

Comment:
initialize with 0 slots & `set` to `input_concurrency` before `enter` function is called maybe

Reply:
Unfortunately in python<=3.9, the semaphore has to be instantiated in the same event loop as the acquire so can't instantiate here / before `enter` is called gives errors